packages/microsoft-reactnative-sampleapps/windows/SampleLibraryCPP/SampleModuleCPP.h

```diff
@@ -75,7 +75,7 @@ struct SampleModuleCppImpl {
   }
 
   REACT_METHOD(ExplicitPromiseMethod);
-  void ExplicitPromiseMethod(winrt::Microsoft::ReactNative::ReactPromise<double> &&result) noexcept {
+  void ExplicitPromiseMethod(winrt::Microsoft::ReactNative::ReactPromise<double> const &result) noexcept {
```

Comment:
I guess I'd missed that winrt types were part of the interface for our nativemodules?  I had thought most of that kind of thing was hidden inside the macros.  It doesn't seem very friendly to an ios/android dev trying to write a xplat C++ module?  Yes I know winrt works on those too, but its likely a big hammer for non-windows centric devs to be taking into their xplat code? Is this the same type thats part of the ReactHost interface for futures?

Reply:
The ReactPromise is the part that 'adapts' ABI safe code for C++ windows users. This is the simplest implementation of a Promise as we can get and we mimic it after the Java ReactPromise.
Currently all our public C++ code uses the winrt::Microsoft::ReactNative namespace. I guess we can change it to something else.
I do not consider this code to be cross-platform.

This type is different from what we use for ReactHost internally.